### PR TITLE
feat: mantine page wrappers + sidebar

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: https://github.com/lightdash/lightdash/issues/4896
Closes: https://github.com/lightdash/lightdash/issues/5332
Closes: https://github.com/lightdash/lightdash/issues/5351

### Description:

- [x] create new page component which abstracts all other page wrappers with simple props
migrates
- [x] migrate old sidebar and use it everywhere

new page styling options
```
    withFixedContent?: boolean;
    withFooter?: boolean;
    withFullHeight?: boolean;
    withHeader?: boolean;
    withNavbar?: boolean;
    withSidebar?: boolean;
    withSidebarFooter?: boolean;
    withPaddedContent?: boolean;
    withCenteredContent?: boolean;
```

### CODE REVIEW

File: packages/frontend/src/pages/CreateProjectSettings.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
-import React, { FC } from 'react';
+import { FC } from 'react';
 import { useQueryClient } from 'react-query';
 import { useHistory, useParams } from 'react-router-dom';
+
```
Comment (joaoviana) [inline packages/frontend/src/pages/CreateProjectSettings.tsx:4]: do you like to have this new line separation between modules and non-module imports?

File: packages/frontend/src/components/ProjectConnection/ProjectConnection.styles.tsx [hunk 2]
```diff
@@ -67,15 +66,15 @@ export const LeftPanelMessage = styled.p`
     color: ${Colors.GRAY1};
 `;
 
+// z-index: ${SIDEBAR_Z_INDEX - 1};
+// padding-left: ${SIDEBAR_WIDTH}px;
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectConnection/ProjectConnection.styles.tsx:70]: perhaps can be removed?

File: packages/frontend/src/components/common/Page/Page.tsx [hunk 3]
```diff
@@ -1,32 +1,170 @@
...
+    withSidebar?: boolean;
+    withSidebarFooter?: boolean;
+    withPaddedContent?: boolean;
+    withCenteredContent?: boolean;
 };
 
-const PageBase = styled('div')<Props>`
-    min-height: calc(
-        100vh - ${(props) => (props.isFullHeight ? '0px' : '50px')}
+const usePageStyles = createStyles<string, StyleProps>((theme, params) => {
```
Comment (joaoviana) [inline packages/frontend/src/components/common/Page/Page.tsx:24]: 👏

**Review by joaoviana — CHANGES_REQUESTED** [review]
wow! a few qs and requests!

File: packages/frontend/src/pages/UserActivity.tsx [hunk 4]
```diff
@@ -144,175 +143,164 @@ const UserActivity: FC = () => {
...
-                            </BigNumberLabel>
-                        </BigNumberContainer>
-                    </ActivityCard>
-                    <ActivityCard grid="editors">
-                        <BigNumberContainer>
-                            <BigNumber>{data.numberEditors}</BigNumber>
-                            <BigNumberLabel>Number of editors</BigNumberLabel>
-                        </BigNumberContainer>
-                    </ActivityCard>
+        <Page title={`User activity for ${project?.name}`}>
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/pages/UserActivity.tsx:146]: no more, Page component takes care of it

File: packages/frontend/src/pages/SqlRunner.tsx [hunk 5]
```diff
@@ -197,12 +192,12 @@ const SqlRunnerPage = () => {
...
 
     return (
-        <PageWithSidebar>
-            <Helmet>
-                <title>SQL Runner - Lightdash</title>
-            </Helmet>
-
-            <Sidebar>
+        <Page
+            title="SQL Runner"
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/pages/SqlRunner.tsx:196]: Inside page

```
            {title ? (
                <Helmet>
                    <title>{title} - Lightdash</title>
                </Helmet>
            ) : null}
```

File: packages/frontend/src/pages/SavedExplorer.tsx [hunk 6]
```diff
@@ -77,62 +63,16 @@ const SavedExplorer = () => {
...
-
-                <PageContentContainer hasDraggableSidebar>
-                    <Explorer />
-                </PageContentContainer>
-            </PageWrapper>
+            <Page
+                title={data?.name}
+                header={<SavedChartsHeader />}
+                sidebar={<ExplorePanel />}
+                sidebarOpened={isEditMode}
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/pages/SavedExplorer.tsx:70]: updated!

File: packages/frontend/src/pages/SavedDashboards.tsx [hunk 7]
```diff
@@ -66,16 +67,10 @@ const SavedDashboards = () => {
...
     return (
-        <Center my="md">
-            <Helmet>
-                <title>Dashboards - Lightdash</title>
-            </Helmet>
-
-            {/* FIXME: use Mantine sizes for width */}
-            <Stack spacing="xl" w={900}>
-                <Group position="apart" mt="xs">
+        <Page title="Dashboards" withFixedContent>
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/pages/SavedDashboards.tsx:70]: all inner pages are width 100%

`withFixedContent` applies this styles so it's covered

```
  marginLeft: 'auto',
  marginRight: 'auto',
  width: 900,
```

File: packages/frontend/src/mantineTheme.ts [hunk 8]
```diff
@@ -85,7 +86,11 @@ export const getMantineThemeOverride = (overrides?: {
         },
     },
 
-    globalStyles: () => ({
+    globalStyles: (theme) => ({
+        'html, body': {
+            backgroundColor: theme.colors.gray[0],
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/mantineTheme.ts:91]: ya, we were applying bg color only on some routes and not on body or html

File: packages/frontend/src/components/ProjectConnection/ProjectConnection.styles.tsx (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/components/ProjectConnection/ProjectConnection.styles.tsx]: all gone!

**Review by joaoviana — APPROVED** [review]
🚢 it!

**Review by joaoviana — APPROVED** [review]

**Pull request merged**